GeometricSeqPhenotype.java

```diff
@@ -0,0 +1,308 @@
+            // get indices for the matrix based on the wild type and mutant amino acids
+            // matrix i,j correspond with the String "ACDEFGHIKLMNPQRSTWYV"
+            int mSiteMutationVectors = Parameters.AlphabetType.AMINO_ACIDS.getValidCharacters().indexOf(wildTypeAminoAcid);
+            int nSiteMutationVectors = Parameters.AlphabetType.AMINO_ACIDS.getValidCharacters().indexOf(mutantAminoAcid);
+
+            double[] mutations = Biology.SiteMutationVectors.VECTORS.getVector(mutationIndexSite, mSiteMutationVectors, nSiteMutationVectors); // use matrix at site # where mutation is occurring
+            mutA = getTraitA() + mutations[0]; // r * cos(theta)
+            mutB = getTraitB() + mutations[1]; // r * sin(theta)
+        }
+
+        return new double[]{mutA, mutB};
+    }
+
+    // Mutates nucleotide sequence at given mutationIndexSiteSequence with given char mutatedNucleotide.
+    // Returns the wild type and mutant amino acid as a String[]
+    //
+    // package private helper method so that the updating of the nucleotide sequence can be tested in TestGeometricSeqPhenotype.java
+    String[] mutateHelper(int mutationIndexSiteSequence, char mutatedNucleotide) {
+        int mutationIndexSiteCodon = mutationIndexSiteSequence % 3; // index of nucleotide being mutated in codon {0, 1, 2}
+        int mutationIndexSite = mutationIndexSiteSequence / 3; // site # where mutation is occurring {0, . . ., total number of sites - 1}
+        int codonStartIndex = 3 * mutationIndexSite; // start index of nucleotide being mutated in codon {0, 3, 6, . . .}
+
+        String originalCodon = "" + this.nucleotideSequence[codonStartIndex] +
+                this.nucleotideSequence[codonStartIndex + 1] +
+                this.nucleotideSequence[codonStartIndex + 2];
+        String wildTypeAminoAcid = Biology.CodonMap.CODONS.getAminoAcid(originalCodon);
+
+        // get new codon after mutation occurs
+        StringBuilder mutatedCodon = new StringBuilder(originalCodon);
+        mutatedCodon.setCharAt(mutationIndexSiteCodon, mutatedNucleotide);
+        String mutantAminoAcid = Biology.CodonMap.CODONS.getAminoAcid(mutatedCodon.toString());
+
+        return new String[]{wildTypeAminoAcid, mutantAminoAcid};
+    }
+
+    /**
+     * Returns the sequence, the position of this GeometricSeqPhenotype, and number of epitope and non-epitope mutations
+     * (i.e. the String representation of the GeometricSeqPhenotype represented by this).
+     * Valid example outputs include "ACG, 0.0, 0.0, 0, 0" and "ACGTGTACGTGT, 2.3, 8.9, 40, 10".
+     *
+     * @return the String representation of the GeometricSeqPhenotype represented by this.
+     */
+    public String toString() {
+        String fullString = String.format("%s, %.4f, %.4f, %d, %d", String.valueOf(this.nucleotideSequence), this.getTraitA(),
+                                          this.getTraitB(), this.eMutation, this.neMutation);
```

Comment:
would this still work if it was called on a GSP object w/o epitope data?

Reply:
Yes, it does. If there are no epitope sites, then only the field, `nE`, will be updated.